review-notifications/src/Github.js

```diff
@@ -1,38 +1,222 @@
+        updated: prObject.updated_at,
+        creator: prObject.user.login,
+        assignees: prObject.assignees.map(assigner => assigner.login),
+        reviewers: prObject.requested_reviewers.map(rev => rev.login),
+        mentioned: mentionedUsers ? mentionedUsers : [],
+      };
+    } catch (error) {
+      console.log(error);
+    }
   }
 
-  handleClick() {
-    axios
-      .get('https://api.github.com/repos/facebook/react/pulls')
-      .then(response => {
-        this.setState({
-          pulls: response.data,
-        });
-      })
-      .catch(error => {
-        console.log(error);
+  filterPullRequests(prData) {
+    let createdPR = [];
+    let assignedPR = [];
+    let mentionedPR = [];
+    let reviewedPR = [];
+    [].concat.apply([], prData).forEach(prObject => {
+      if (prObject.creator === this.state.user) {
+        this.addPRToList(createdPR, prObject);
+      }
+      prObject.assignees.forEach(assigner => {
+        if (assigner === this.state.user)
+          this.addPRToList(assignedPR, prObject);
+      });
+      prObject.mentioned.forEach(mentionedUser => {
+        if (mentionedUser === this.state.user)
+          this.addPRToList(mentionedPR, prObject);
+      });
+      prObject.reviewers.forEach(reviewer => {
+        if (reviewer === this.state.user)
+          this.addPRToList(reviewedPR, prObject);
       });
+    });
+    this.setState({
+      createdPR: createdPR,
+      assignedPR: assignedPR,
+      mentionedPR: mentionedPR,
+      reviewedPR: reviewedPR,
+    });
   }
-  listOfTitles() {
-    const list = this.state.pulls.map(pr => <li key={pr.id}>{pr.title}</li>);
-    return <ul>{list}</ul>;
+
+  async getPullRequests() {
+    let query = `https://api.github.com/users/${this.state.user}/repos`;
+    if (this.state.token !== '' && this.state.auth) {
+      query = `https://api.github.com/user/repos?access_token=${
+        this.state.token
+      }`;
+    }
+    let prLinksList = [];
+    try {
+      const response = await axios.get(query);
+      prLinksList = response.data.map(repo =>
+        repo.pulls_url.replace('{/number}', '')
+      );
+    } catch (error) {
+      console.log(error);
+    }
+    try {
+      const pullRequests = await axios.all(
+        prLinksList.map(prLink =>
+          axios.get(`${prLink}?access_token=${this.state.token}`)
+        )
+      );
+      const prData = await Promise.all(
+        pullRequests.map(
+          async prList =>
+            await Promise.all(
+              prList.data.map(
+                async pullRequest => await this.extractDataFromPR(pullRequest)
+              )
+            )
+        )
+      );
+      this.filterPullRequests(prData);
+      this.setState({ haveData: true });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  listOfPullRequest() {
+    let list = [];
+    this.state.prOptions.forEach(option => {
+      if (option.isChecked)
+        switch (option.value) {
+          case 'Created':
```

Comment:
You might wanna consider exporting these case values to an `enum` since you reuse it in a few places. Prevents bugs from typos